logstash-core/src/main/java/org/logstash/config/source/Base.java

```diff
@@ -0,0 +1,145 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.logstash.config.source;
+
+import org.jruby.RubyArray;
+import org.jruby.RubyObject;
+import org.jruby.RubyString;
+import org.jruby.runtime.builtin.IRubyObject;
+import org.logstash.config.ir.PipelineConfig;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.logstash.RubyUtil.RUBY;
+
+public class Base {
+    protected RubyObject settings;
+    protected List<String> conflictMessages = new ArrayList<>();
+
+    public Base(RubyObject logstashSettings) {
+        settings = logstashSettings;
+    }
+
+    public List<String> getConflictMessages() {
+        return conflictMessages;
+    }
+
+    // this must be used when in Ruby code was @settings = <new value>
+    public void updateSettings(RubyObject settings) {
+        this.settings = settings;
+    }
+
+    public List<PipelineConfig> pipelineConfigs() {
+        throw new UnsupportedOperationException("`pipelineConfigs` must be implemented!");
+    }
+
+    public boolean isMatch() {
+        throw new UnsupportedOperationException("`isMatch` must be implemented!");
+    }
+
+    public boolean isConfigConflict() {
+        throw new UnsupportedOperationException("`isConfigConflict` must be implemented!");
+    }
+
+    //return subclass of LogStash::Setting
+    public IRubyObject configReloadAutomaticSetting() {
+        return this.settings.callMethod(RUBY.getCurrentContext(), "get_setting",
+                RubyString.newString(RUBY, "config.reload.automatic"));
+    }
+
+    public boolean configReloadAutomatic() {
+        final IRubyObject value = configReloadAutomaticSetting().callMethod(RUBY.getCurrentContext(), "value");
+        return value.toJava(Boolean.class);
+    }
+
+    public boolean isConfigReloadAutomatic() {
+        final IRubyObject valueIsSet = configReloadAutomaticSetting().callMethod(RUBY.getCurrentContext(), "set?");
+        return valueIsSet.toJava(Boolean.class);
+    }
+
+    //return subclass of LogStash::Setting
+    public IRubyObject configStringSetting() {
+        return this.settings.callMethod(RUBY.getCurrentContext(), "get_setting",
+                RubyString.newString(RUBY, "config.string"));
+    }
+
+    public String configString() {
+        final IRubyObject setting = configStringSetting();
+        final IRubyObject value = setting.callMethod(RUBY.getCurrentContext(), "value");
+        return value.toJava(String.class);
+    }
+
+    public boolean isConfigString() {
+        return configString() != null;
+    }
+
+    // return subclass of LogStash::Setting
+    public IRubyObject configPathSetting() {
+        return this.settings.callMethod(RUBY.getCurrentContext(), "get_setting",
+                RubyString.newString(RUBY, "path.config"));
+    }
+
+    public String configPath() {
+        return configPathSetting().callMethod(RUBY.getCurrentContext(), "value").toJava(String.class);
```

Comment:
same as above 2 dynamic `callMethods` than converting Ruby String -> Java and back to Ruby. 
makes me wonder how often these are used - it's likely negligible but still worth an inspection (they're probably used only during boot and we do not want to slow-down startup even more).